### DISCUSSION TITLE
call back for event loop is ready

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#17049] [Q&A]
Hi,
I want to initialize my App after the all worker threads are initialized and ready.
thus, the main function is not a good place for that.
document onload is also not an option, because it is called before the main function.
is there any event  that I can get for that ?

**Comment by sbc100** [comment]
Does this work: https://emscripten.org/docs/getting_started/FAQ.html#how-can-i-tell-when-the-page-is-fully-loaded-and-it-is-safe-to-call-compiled-functions

**Comment by aharondavid** [comment]
Thanks a lot

**Comment by aharondavid** [comment]
In the link above there are two option and none of them can tell us that all worker threads are loaded (checked in debug):
1. 
 ```
int main() {
  EM_ASM( allReady() );
}
```
according to https://github.com/emscripten-core/emscripten/issues/16763, we should not create a pthread inside the main, because
it will not attached to the existing worker thread. but `allReady()` is called from the main, 

2. 
  ```
 onRuntimeInitialized: function() {
      Module._foobar(); // foobar was exported
    }
```

the  onRuntimeInitialized is called before the main is called.

so, the question is still relevant.

**Comment by sbc100** [comment]
When you say "all worker threads are loaded", are you talking about thread in `PTHREAD_POOL_SIZE`?  If so then I believe they are all started before `main` or `onRuntimeInitialized` is called.   You should be free to start threads as soon as `main` or `onRuntimeInitialized` is called.   Are you experiencing problems starting threads at the point?  Do you have an example you could share?

**Comment by aharondavid** [comment]
yes, I mean to PTHREAD_POOL_SIZE,  because some limitations, I need to avoid from doing your suggest in https://github.com/emscripten-core/emscripten/issues/16763#issuecomment-1104347063, I thought to get over on it by such event. when I will get that event, I will execute the On_Press_UI_Buton describe here https://github.com/emscripten-core/emscripten/issues/16763#issuecomment-1104508429
